File: lib/src/screens/settings_support_screen.dart
import 'package:flutter/material.dart';
import 'package:url_launcher/url_launcher.dart';
import '../constants/app_color.dart';

class SettingsSupportScreen extends StatefulWidget {
  const SettingsSupportScreen({Key? key}) : super(key: key);

  @override
  State<SettingsSupportScreen> createState() => _SettingsSupportScreenState();
}

class _SettingsSupportScreenState extends State<SettingsSupportScreen> {
  // Animation states
  bool _showContent = false;
  bool _showSuccessDialog = false;
  String _dialogMessage = "";

  // Settings states
  bool _isNotificationsEnabled = true;
  bool _isLocationEnabled = true;
  bool _isReminderEnabled = true;
  bool _isDarkMode = false;
  bool _isAccessibilityEnabled = false;
  bool _isBiometricEnabled = false;
  String _measurementSystem = "Imperial";
  String _languagePreference = "English";

  // Measurement system options
  final List<String> _measurementOptions = ["Imperial", "Metric"];

  // Language options
  final List<String> _languageOptions = [
    "English",
    "Spanish",
    "French",
    "German",
    "Chinese"
  ];

  @override
  void initState() {
    super.initState();

    // Trigger animation after a delay
    Future.delayed(const Duration(milliseconds: 100), () {
      if (mounted) {
        setState(() {
          _showContent = true;
        });
      }
    });
  }

  void _showSuccess(String message) {
    setState(() {
      _showSuccessDialog = true;
      _dialogMessage = message;
    });

    // Hide dialog after a delay
    Future.delayed(const Duration(seconds: 2), () {
      if (mounted) {
        setState(() {
          _showSuccessDialog = false;
          _dialogMessage = "";
        });
      }
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text(
          'Settings & Support',
          style: TextStyle(
            fontSize: 24,
            fontWeight: FontWeight.w600,
            color: AppColor.textDarkBlue,
          ),
        ),
        backgroundColor: Colors.transparent,
        elevation: 0,
        iconTheme: const IconThemeData(color: AppColor.textDarkBlue),
      ),
      body: Stack(
        children: [
          Container(
            color: AppColor.backgroundGray,
            child: AnimatedOpacity(
              opacity: _showContent ? 1.0 : 0.0,
              duration: const Duration(milliseconds: 500),
              curve: Curves.easeOut,
              child: AnimatedSlide(
                offset: _showContent ? Offset.zero : const Offset(0, 0.05),
                duration: const Duration(milliseconds: 500),
                curve: Curves.easeOutCubic,
                child: SingleChildScrollView(
                  padding: const EdgeInsets.all(16),
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      // Notification Settings Card
                      _buildSettingsCard(
                        title: "Notifications",
                        icon: Icons.notifications_outlined,
                        iconTint: AppColor.primaryBlue,
                        children: [
                          _buildSwitchItem(
                            title: "Enable Notifications",
                            description:
                                "Receive important updates about your rides",
                            value: _isNotificationsEnabled,
                            onChanged: (value) {
                              setState(() {
                                _isNotificationsEnabled = value;
                              });
                            },
                          ),
                          _buildSwitchItem(
                            title: "Ride Reminders",
                            description:
                                "Get reminded 1 hour before scheduled rides",
                            value: _isReminderEnabled,
                            onChanged: (value) {
                              setState(() {
                                _isReminderEnabled = value;
                              });
                            },
                            enabled: _isNotificationsEnabled,
                          ),
                          _buildSwitchItem(
                            title: "Driver Updates",
                            description:
                                "Get notified when your driver is assigned or nearby",
                            value: _isNotificationsEnabled,
                            onChanged: (value) {
                              // This is controlled by the main notification toggle
                            },
                            enabled: _isNotificationsEnabled,
                            isLast: true,
                          ),
                        ],
                      ),

                      const SizedBox(height: 16),

                      // Privacy & Security Card
                      _buildSettingsCard(
                        title: "Privacy & Security",
                        icon: Icons.security_outlined,
                        iconTint: AppColor.primaryPurple,
                        children: [
                          _buildSwitchItem(
                            title: "Location Services",
                            description:
                                "Allow the app to access your location",
                            value: _isLocationEnabled,
                            onChanged: (value) {
                              setState(() {
                                _isLocationEnabled = value;
                              });
                            },
                          ),
                          _buildSwitchItem(
                            title: "Biometric Authentication",
                            description:
                                "Use fingerprint or face ID to sign in",
                            value: _isBiometricEnabled,
                            onChanged: (value) {
                              setState(() {
                                _isBiometricEnabled = value;
                              });
                            },
                          ),
                          _buildClickableItem(
                            title: "Privacy Policy",
                            description: "Read about how we handle your data",
                            onTap: () async {
                              final Uri url =
                                  Uri.parse('https://example.com/privacy');
                              if (await canLaunchUrl(url)) {
                                await launchUrl(url);
                              }
                            },
                            isLast: true,
                          ),
                        ],
                      ),

                      const SizedBox(height: 16),

                      // Appearance Card
                      _buildSettingsCard(
                        title: "Appearance",
                        icon: Icons.palette_outlined,
                        iconTint: const Color(0xFF4EA8DE), // Light blue
                        children: [
                          _buildSwitchItem(
                            title: "Dark Mode",
                            description: "Switch between light and dark themes",
                            value: _isDarkMode,
                            onChanged: (value) {
                              setState(() {
                                _isDarkMode = value;
                              });
                            },
                          ),
                          _buildSwitchItem(
                            title: "Accessibility Features",
                            description:
                                "Larger text, increased contrast, etc.",
                            value: _isAccessibilityEnabled,
                            onChanged: (value) {
                              setState(() {
                                _isAccessibilityEnabled = value;
                              });
                            },
                            isLast: true,
                          ),
                        ],
                      ),

                      const SizedBox(height: 16),

                      // Preferences Card
                      _buildSettingsCard(
                        title: "Preferences",
                        icon: Icons.tune_outlined,
                        iconTint: const Color(0xFFE76F51), // Orange
                        children: [
                          _buildDropdownItem(
                            title: "Language",
                            description: "Choose your preferred language",
                            value: _languagePreference,
                            options: _languageOptions,
                            onChanged: (value) {
                              if (value != null) {
                                setState(() {
                                  _languagePreference = value;
                                });
                              }
                            },
                          ),
                          _buildDropdownItem(
                            title: "Measurement System",
                            description:
                                "Choose between Imperial and Metric units",
                            value: _measurementSystem,
                            options: _measurementOptions,
                            onChanged: (value) {
                              if (value != null) {
                                setState(() {
                                  _measurementSystem = value;
                                });
                              }
                            },
                            isLast: true,
                          ),
                        ],
                      ),

                      const SizedBox(height: 16),

                      // Support Card
                      _buildSettingsCard(
                        title: "Help & Support",
                        icon: Icons.help_outline,
                        iconTint: const Color(0xFF38B000), // Green
                        children: [
                          _buildClickableItem(
                            title: "Contact Support",
                            description: "Get help with your account or rides",
                            onTap: () async {
                              final Uri emailUri = Uri(
                                scheme: 'mailto',
                                path: 'support@healthride.com',
                                queryParameters: {'subject': 'Support Request'},
                              );
                              if (await canLaunchUrl(emailUri)) {
                                await launchUrl(emailUri);
                              }
                            },
                          ),
                          _buildClickableItem(
                            title: "FAQs",
                            description: "Find answers to common questions",
                            onTap: () async {
                              final Uri url =
                                  Uri.parse('https://example.com/faq');
                              if (await canLaunchUrl(url)) {
                                await launchUrl(url);
                              }
                            },
                          ),
                          _buildClickableItem(
                            title: "Report a Problem",
                            description:
                                "Let us know if something isn't working",
                            onTap: () {
                              _showSuccess(
                                  "Problem reported successfully. We'll get back to you soon.");
                            },
                            isLast: true,
                          ),
                        ],
                      ),

                      const SizedBox(height: 16),

                      // About App Card
                      _buildSettingsCard(
                        title: "About",
                        icon: Icons.info_outline,
                        iconTint: AppColor.textDarkBlue,
                        children: [
                          _buildClickableItem(
                            title: "Version",
                            description: "1.0.0 (build 100)",
                            onTap: () {
                              // Perhaps show a dialog with version details
                            },
                          ),
                          _buildClickableItem(
                            title: "Terms of Service",
                            description: "Read our terms and conditions",
                            onTap: () async {
                              final Uri url =
                                  Uri.parse('https://example.com/terms');
                              if (await canLaunchUrl(url)) {
                                await launchUrl(url);
                              }
                            },
                          ),
                          _buildClickableItem(
                            title: "Open Source Licenses",
                            description: "View third-party licenses",
                            onTap: () {
                              // Show licenses page
                            },
                            isLast: true,
                          ),
                        ],
                      ),

                      const SizedBox(height: 24),
                    ],
                  ),
                ),
              ),
            ),
          ),

          // Success Dialog
          if (_showSuccessDialog)
            Center(
              child: Container(
                margin: const EdgeInsets.all(32),
                padding: const EdgeInsets.all(20),
                decoration: BoxDecoration(
                  color: Colors.white,
                  borderRadius: BorderRadius.circular(16),
                  boxShadow: [
                    BoxShadow(
                      color: Colors.black.withOpacity(0.1),
                      blurRadius: 16,
                      offset: const Offset(0, 4),
                    ),
                  ],
                ),
                child: Column(
                  mainAxisSize: MainAxisSize.min,
                  children: [
                    Container(
                      width: 60,
                      height: 60,
                      decoration: BoxDecoration(
                        color: AppColor.successGreen.withOpacity(0.1),
                        shape: BoxShape.circle,
                      ),
                      child: const Icon(
                        Icons.check_circle,
                        size: 32,
                        color: AppColor.successGreen,
                      ),
                    ),
                    const SizedBox(height: 16),
                    Text(
                      _dialogMessage,
                      textAlign: TextAlign.center,
                      style: const TextStyle(
                        fontSize: 16,
                        fontWeight: FontWeight.w500,
                        color: AppColor.textDarkBlue,
                      ),
                    ),
                  ],
                ),
              ),
            ),
        ],
      ),
    );
  }

  Widget _buildSettingsCard({
    required String title,
    required IconData icon,
    required Color iconTint,
    required List<Widget> children,
  }) {
    return Card(
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(16),
      ),
      elevation: 2,
      child: Padding(
        padding: const EdgeInsets.all(20),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Row(
              children: [
                Container(
                  width: 36,
                  height: 36,
                  decoration: BoxDecoration(
                    color: iconTint.withOpacity(0.1),
                    shape: BoxShape.circle,
                  ),
                  child: Icon(
                    icon,
                    size: 20,
                    color: iconTint,
                  ),
                ),
                const SizedBox(width: 12),
                Text(
                  title,
                  style: const TextStyle(
                    fontSize: 18,
                    fontWeight: FontWeight.bold,
                    color: AppColor.textDarkBlue,
                  ),
                ),
              ],
            ),
            const SizedBox(height: 16),
            const Divider(color: Color(0xFFF0F4F8)),
            const SizedBox(height: 16),
            ...children,
          ],
        ),
      ),
    );
  }

  Widget _buildSwitchItem({
    required String title,
    required String description,
    required bool value,
    required ValueChanged<bool> onChanged,
    bool enabled = true,
    bool isLast = false,
  }) {
    return Column(
      children: [
        Row(
          children: [
            Expanded(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Text(
                    title,
                    style: TextStyle(
                      fontSize: 16,
                      fontWeight: FontWeight.w500,
                      color: enabled
                          ? AppColor.textDarkBlue
                          : AppColor.textLightGray,
                    ),
                  ),
                  const SizedBox(height: 4),
                  Text(
                    description,
                    style: TextStyle(
                      fontSize: 14,
                      color: enabled
                          ? AppColor.textMediumGray
                          : AppColor.textLightGray,
                    ),
                  ),
                ],
              ),
            ),
            Switch(
              value: value,
              onChanged: enabled ? onChanged : null,
              activeColor: AppColor.primaryBlue,
              trackColor: MaterialStateProperty.resolveWith((states) {
                if (states.contains(MaterialState.disabled)) {
                  return AppColor.textLightGray.withOpacity(0.3);
                }
                return value ? AppColor.primaryBlue.withOpacity(0.3) : null;
              }),
            ),
          ],
        ),
        if (!isLast) ...[
          const SizedBox(height: 16),
          const Divider(color: Color(0xFFF0F4F8)),
          const SizedBox(height: 16),
        ],
      ],
    );
  }

  Widget _buildClickableItem({
    required String title,
    required String description,
    required VoidCallback onTap,
    bool isLast = false,
  }) {
    return Column(
      children: [
        InkWell(
          onTap: onTap,
          borderRadius: BorderRadius.circular(8),
          child: Padding(
            padding: const EdgeInsets.symmetric(vertical: 8),
            child: Row(
              children: [
                Expanded(
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Text(
                        title,
                        style: const TextStyle(
                          fontSize: 16,
                          fontWeight: FontWeight.w500,
                          color: AppColor.textDarkBlue,
                        ),
                      ),
                      const SizedBox(height: 4),
                      Text(
                        description,
                        style: const TextStyle(
                          fontSize: 14,
                          color: AppColor.textMediumGray,
                        ),
                      ),
                    ],
                  ),
                ),
                const Icon(
                  Icons.arrow_forward_ios,
                  size: 16,
                  color: AppColor.textLightGray,
                ),
              ],
            ),
          ),
        ),
        if (!isLast) ...[
          const SizedBox(height: 16),
          const Divider(color: Color(0xFFF0F4F8)),
          const SizedBox(height: 16),
        ],
      ],
    );
  }

  Widget _buildDropdownItem({
    required String title,
    required String description,
    required String value,
    required List<String> options,
    required ValueChanged<String?> onChanged,
    bool isLast = false,
  }) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Text(
          title,
          style: const TextStyle(
            fontSize: 16,
            fontWeight: FontWeight.w500,
            color: AppColor.textDarkBlue,
          ),
        ),
        const SizedBox(height: 4),
        Text(
          description,
          style: const TextStyle(
            fontSize: 14,
            color: AppColor.textMediumGray,
          ),
        ),
        const SizedBox(height: 12),
        Container(
          width: double.infinity,
          padding: const EdgeInsets.symmetric(horizontal: 16),
          decoration: BoxDecoration(
            borderRadius: BorderRadius.circular(8),
            border: Border.all(color: const Color(0xFFE5E7EB)),
            color: Colors.white,
          ),
          child: DropdownButton<String>(
            value: value,
            onChanged: onChanged,
            items: options.map((option) {
              return DropdownMenuItem<String>(
                value: option,
                child: Text(option),
              );
            }).toList(),
            underline: const SizedBox(), // Remove underline
            isExpanded: true,
            icon: const Icon(Icons.keyboard_arrow_down,
                color: AppColor.textMediumGray),
          ),
        ),
        if (!isLast) ...[
          const SizedBox(height: 16),
          const Divider(color: Color(0xFFF0F4F8)),
          const SizedBox(height: 16),
        ],
      ],
    );
  }
}

File: backup/kotlin/ui/SettingsSupportScreen.kt
package com.example.healthride.ui

import android.content.Intent
import android.net.Uri
import android.widget.Toast
import androidx.compose.animation.*
import androidx.compose.animation.core.*
import androidx.compose.foundation.*
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.*
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material.icons.outlined.*
import androidx.compose.material.icons.rounded.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.*
import androidx.compose.ui.graphics.*
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.window.Dialog
import com.example.healthride.ui.theme.*
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun SettingsSupportScreen(
    onBackClick: () -> Unit = {}
) {
    val context = LocalContext.current
    val scrollState = rememberScrollState()
    val coroutineScope = rememberCoroutineScope()

    // Animation states
    var showContent by remember { mutableStateOf(false) }
    var showSuccessDialog by remember { mutableStateOf(false) }
    var dialogMessage by remember { mutableStateOf("") }

    // Settings states
    var isNotificationsEnabled by remember { mutableStateOf(true) }
    var isLocationEnabled by remember { mutableStateOf(true) }
    var isReminderEnabled by remember { mutableStateOf(true) }
    var isDarkMode by remember { mutableStateOf(false) }
    var isAccessibilityEnabled by remember { mutableStateOf(false) }
    var isBiometricEnabled by remember { mutableStateOf(false) }
    var measurementSystem by remember { mutableStateOf("Imperial") }
    var languagePreference by remember { mutableStateOf("English") }

    // Trigger animations
    LaunchedEffect(Unit) {
        delay(100)
        showContent = true
    }

    Scaffold(
        topBar = {
            TopAppBar(
                title = {
                    Text(
                        "Settings & Support",
                        fontWeight = FontWeight.SemiBold
                    )
                },
                navigationIcon = {
                    IconButton(onClick = onBackClick) {
                        Icon(
                            imageVector = Icons.Default.ArrowBack,
                            contentDescription = "Back",
                            tint = TextDarkBlue
                        )
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = Color.Transparent,
                    titleContentColor = TextDarkBlue
                )
            )
        }
    ) { paddingValues ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .background(BackgroundGray)
                .padding(paddingValues)
                .verticalScroll(scrollState)
                .padding(horizontal = 16.dp, vertical = 8.dp)
        ) {
            AnimatedVisibility(
                visible = showContent,
                enter = fadeIn() + slideInVertically(initialOffsetY = { it / 4 })
            ) {
                Column(
                    verticalArrangement = Arrangement.spacedBy(16.dp)
                ) {
                    // Notification Settings Card
                    SettingsCard(
                        title = "Notifications",
                        icon = Icons.Outlined.Notifications,
                        iconTint = PrimaryBlue
                    ) {
                        SwitchSettingItem(
                            title = "Enable Notifications",
                            description = "Receive important updates about your rides",
                            checked = isNotificationsEnabled,
                            onCheckedChange = { isNotificationsEnabled = it }
                        )

                        SwitchSettingItem(
                            title = "Ride Reminders",
                            description = "Get reminded 1 hour before scheduled rides",
                            checked = isReminderEnabled,
                            onCheckedChange = { isReminderEnabled = it },
                            enabled = isNotificationsEnabled
                        )

                        SwitchSettingItem(
                            title = "Driver Updates",
                            description = "Get notified when your driver is assigned or nearby",
                            checked = isNotificationsEnabled,
                            onCheckedChange = { /* This is controlled by the main notification toggle */ },
                            enabled = isNotificationsEnabled,
                            isLast = true
                        )
                    }

                    // Privacy & Security Card
                    SettingsCard(
                        title = "Privacy & Security",
                        icon = Icons.Outlined.Security,
                        iconTint = PrimaryPurple
                    ) {
                        SwitchSettingItem(
                            title = "Location Services",
                            description = "Allow the app to access your location",
                            checked = isLocationEnabled,
                            onCheckedChange = { isLocationEnabled = it }
                        )

                        SwitchSettingItem(
                            title = "Biometric Authentication",
                            description = "Use fingerprint or face ID to sign in",
                            checked = isBiometricEnabled,
                            onCheckedChange = { isBiometricEnabled = it }
                        )

                        ClickableSettingItem(
                            title = "Privacy Policy",
                            description = "Read about how we handle your data",
                            onClick = {
                                val intent = Intent(Intent.ACTION_VIEW, Uri.parse("https://example.com/privacy"))
                                context.startActivity(intent)
                            },
                            isLast = true
                        )
                    }

                    // Appearance Card
                    SettingsCard(
                        title = "Appearance",
                        icon = Icons.Outlined.Palette,
                        iconTint = Color(0xFF4EA8DE) // Light blue
                    ) {
                        SwitchSettingItem(
                            title = "Dark Mode",
                            description = "Switch between light and dark themes",
                            checked = isDarkMode,
                            onCheckedChange = { isDarkMode = it }
                        )

                        SwitchSettingItem(
                            title = "Accessibility Features",
                            description = "Larger text, increased contrast, etc.",
                            checked = isAccessibilityEnabled,
                            onCheckedChange = { isAccessibilityEnabled = it },
                            isLast = true
                        )
                    }

                    // Preferences Card
                    SettingsCard(
                        title = "Preferences",
                        icon = Icons.Outlined.Tune,
                        iconTint = Color(0xFFE76F51) // Orange
                    ) {
                        DropdownSettingItem(
                            title = "Language",
                            description = "Choose your preferred language",
                            currentValue = languagePreference,
                            options = listOf("English", "Spanish", "French", "German", "Chinese"),
                            onValueChange = { languagePreference = it }
                        )

                        DropdownSettingItem(
                            title = "Measurement System",
                            description = "Distance and other measurements",
                            currentValue = measurementSystem,
                            options = listOf("Imperial", "Metric"),
                            onValueChange = { measurementSystem = it },
                            isLast = true
                        )
                    }

                    // Help & Support Card
                    SettingsCard(
                        title = "Help & Support",
                        icon = Icons.Outlined.ContactSupport,
                        iconTint = SuccessGreen
                    ) {
                        ClickableSettingItem(
                            title = "FAQs",
                            description = "Find answers to common questions",
                            onClick = {
                                dialogMessage = "FAQs would open here"
                                showSuccessDialog = true
                            }
                        )

                        ClickableSettingItem(
                            title = "Contact Support",
                            description = "Get help from our team",
                            onClick = {
                                try {
                                    val intent = Intent(Intent.ACTION_SENDTO).apply {
                                        data = Uri.parse("mailto:support@healthride.com")
                                        putExtra(Intent.EXTRA_SUBJECT, "HealthRide Support Request")
                                    }
                                    context.startActivity(intent)
                                } catch (e: Exception) {
                                    Toast.makeText(context, "Could not open email app", Toast.LENGTH_SHORT).show()
                                }
                            }
                        )

                        ClickableSettingItem(
                            title = "Live Chat",
                            description = "Chat with our support team",
                            onClick = {
                                dialogMessage = "Live chat would open here"
                                showSuccessDialog = true
                            },
                            isLast = true
                        )
                    }

                    // About Card
                    SettingsCard(
                        title = "About",
                        icon = Icons.Outlined.Info,
                        iconTint = PrimaryBlue
                    ) {
                        ClickableSettingItem(
                            title = "App Version",
                            description = "1.0.0 (Build 1234)",
                            onClick = { /* No action */ },
                            showArrow = false
                        )

                        ClickableSettingItem(
                            title = "Terms of Service",
                            description = "Read the terms of service",
                            onClick = {
                                val intent = Intent(Intent.ACTION_VIEW, Uri.parse("https://example.com/terms"))
                                context.startActivity(intent)
                            }
                        )

                        ClickableSettingItem(
                            title = "Open Source Licenses",
                            description = "View licenses for third-party software",
                            onClick = {
                                dialogMessage = "Licenses would show here"
                                showSuccessDialog = true
                            },
                            isLast = true
                        )
                    }

                    // Delete Account Button (at the bottom)
                    Button(
                        onClick = {
                            dialogMessage = "Account deletion would be processed here"
                            showSuccessDialog = true
                        },
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(vertical = 16.dp)
                            .height(56.dp),
                        shape = RoundedCornerShape(28.dp),
                        colors = ButtonDefaults.buttonColors(
                            containerColor = ErrorRed.copy(alpha = 0.9f)
                        )
                    ) {
                        Icon(
                            imageVector = Icons.Outlined.DeleteForever,
                            contentDescription = null,
                            modifier = Modifier.size(20.dp)
                        )

                        Spacer(modifier = Modifier.width(8.dp))

                        Text(
                            text = "Delete Account",
                            fontWeight = FontWeight.Bold
                        )
                    }

                    Spacer(modifier = Modifier.height(24.dp))
                }
            }
        }
    }

    // Success Dialog
    if (showSuccessDialog) {
        AlertDialog(
            onDismissRequest = { showSuccessDialog = false },
            shape = RoundedCornerShape(24.dp),
            containerColor = Color.White,
            title = {
                Text(
                    text = "Information",
                    style = MaterialTheme.typography.titleLarge,
                    fontWeight = FontWeight.SemiBold,
                    color = TextDarkBlue,
                    textAlign = TextAlign.Center,
                    modifier = Modifier.fillMaxWidth()
                )
            },
            text = {
                Text(
                    text = dialogMessage,
                    style = MaterialTheme.typography.bodyLarge,
                    color = TextMediumGray,
                    textAlign = TextAlign.Center
                )
            },
            confirmButton = {
                Button(
                    onClick = { showSuccessDialog = false },
                    modifier = Modifier.fillMaxWidth(),
                    shape = RoundedCornerShape(50.dp),
                    colors = ButtonDefaults.buttonColors(
                        containerColor = PrimaryBlue
                    )
                ) {
                    Text("OK")
                }
            }
        )
    }
}

@Composable
fun SettingsCard(
    title: String,
    icon: ImageVector,
    iconTint: Color,
    content: @Composable () -> Unit
) {
    var isExpanded by remember { mutableStateOf(true) }

    Card(
        modifier = Modifier
            .fillMaxWidth()
            .animateContentSize(
                animationSpec = spring(
                    dampingRatio = Spring.DampingRatioMediumBouncy,
                    stiffness = Spring.StiffnessLow
                )
            ),
        shape = RoundedCornerShape(16.dp),
        colors = CardDefaults.cardColors(
            containerColor = SurfaceWhite
        ),
        elevation = CardDefaults.cardElevation(
            defaultElevation = 2.dp
        )
    ) {
        Column(
            modifier = Modifier.fillMaxWidth()
        ) {
            // Card header with expand/collapse
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .clickable { isExpanded = !isExpanded }
                    .padding(horizontal = 20.dp, vertical = 16.dp),
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.SpaceBetween
            ) {
                Row(
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    // Icon with background
                    Box(
                        modifier = Modifier
                            .size(40.dp)
                            .clip(CircleShape)
                            .background(iconTint.copy(alpha = 0.1f)),
                        contentAlignment = Alignment.Center
                    ) {
                        Icon(
                            imageVector = icon,
                            contentDescription = null,
                            tint = iconTint,
                            modifier = Modifier.size(24.dp)
                        )
                    }

                    Spacer(modifier = Modifier.width(16.dp))

                    Text(
                        text = title,
                        style = MaterialTheme.typography.titleMedium,
                        fontWeight = FontWeight.Bold,
                        color = TextDarkBlue
                    )
                }

                // Expand/collapse icon
                val rotation = animateFloatAsState(
                    targetValue = if (isExpanded) 0f else 180f,
                    label = "arrow_rotation"
                )

                Icon(
                    imageVector = Icons.Filled.KeyboardArrowUp,
                    contentDescription = if (isExpanded) "Collapse" else "Expand",
                    tint = TextMediumGray,
                    modifier = Modifier.graphicsLayer { rotationZ = rotation.value }
                )
            }

            // Divider
            Divider(color = Color(0xFFF0F4F8))

            // Card content (collapsible)
            AnimatedVisibility(
                visible = isExpanded,
                enter = expandVertically() + fadeIn(),
                exit = shrinkVertically() + fadeOut()
            ) {
                Column(
                    modifier = Modifier.padding(20.dp)
                ) {
                    content()
                }
            }
        }
    }
}

@Composable
fun SwitchSettingItem(
    title: String,
    description: String,
    checked: Boolean,
    onCheckedChange: (Boolean) -> Unit,
    enabled: Boolean = true,
    isLast: Boolean = false
) {
    Column {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(vertical = 8.dp)
                .alpha(if (enabled) 1f else 0.6f),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.SpaceBetween
        ) {
            Column(
                modifier = Modifier.weight(1f)
            ) {
                Text(
                    text = title,
                    style = MaterialTheme.typography.bodyLarge,
                    fontWeight = FontWeight.Medium,
                    color = TextDarkBlue
                )

                Text(
                    text = description,
                    style = MaterialTheme.typography.bodySmall,
                    color = TextMediumGray
                )
            }

            Spacer(modifier = Modifier.width(16.dp))

            Switch(
                checked = checked,
                onCheckedChange = { if (enabled) onCheckedChange(it) },
                enabled = enabled,
                colors = SwitchDefaults.colors(
                    checkedThumbColor = Color.White,
                    checkedTrackColor = PrimaryBlue,
                    checkedBorderColor = PrimaryBlue,
                    uncheckedThumbColor = Color.White,
                    uncheckedTrackColor = Color.LightGray.copy(alpha = 0.6f),
                    uncheckedBorderColor = Color.LightGray.copy(alpha = 0.6f)
                )
            )
        }

        if (!isLast) {
            Divider(
                color = Color(0xFFF1F5F9),
                modifier = Modifier.padding(vertical = 8.dp)
            )
        }
    }
}

@Composable
fun ClickableSettingItem(
    title: String,
    description: String,
    onClick: () -> Unit,
    isLast: Boolean = false,
    showArrow: Boolean = true
) {
    Column {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .clickable(onClick = onClick)
                .padding(vertical = 8.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.SpaceBetween
        ) {
            Column(
                modifier = Modifier.weight(1f)
            ) {
                Text(
                    text = title,
                    style = MaterialTheme.typography.bodyLarge,
                    fontWeight = FontWeight.Medium,
                    color = TextDarkBlue
                )

                Text(
                    text = description,
                    style = MaterialTheme.typography.bodySmall,
                    color = TextMediumGray
                )
            }

            if (showArrow) {
                Icon(
                    imageVector = Icons.Filled.ChevronRight,
                    contentDescription = "Open",
                    tint = TextLightGray,
                    modifier = Modifier.size(20.dp)
                )
            }
        }

        if (!isLast) {
            Divider(
                color = Color(0xFFF1F5F9),
                modifier = Modifier.padding(vertical = 8.dp)
            )
        }
    }
}

@Composable
fun DropdownSettingItem(
    title: String,
    description: String,
    currentValue: String,
    options: List<String>,
    onValueChange: (String) -> Unit,
    isLast: Boolean = false
) {
    var expanded by remember { mutableStateOf(false) }

    Column {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .clickable { expanded = true }
                .padding(vertical = 8.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.SpaceBetween
        ) {
            Column(
                modifier = Modifier.weight(1f)
            ) {
                Text(
                    text = title,
                    style = MaterialTheme.typography.bodyLarge,
                    fontWeight = FontWeight.Medium,
                    color = TextDarkBlue
                )

                Text(
                    text = description,
                    style = MaterialTheme.typography.bodySmall,
                    color = TextMediumGray
                )
            }

            Spacer(modifier = Modifier.width(8.dp))

            Row(
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = currentValue,
                    style = MaterialTheme.typography.bodyMedium,
                    color = PrimaryBlue,
                    fontWeight = FontWeight.Medium
                )

                Icon(
                    imageVector = Icons.Filled.ArrowDropDown,
                    contentDescription = "Select",
                    tint = PrimaryBlue
                )
            }

            DropdownMenu(
                expanded = expanded,
                onDismissRequest = { expanded = false },
                modifier = Modifier
                    .background(SurfaceWhite)
                    .width(200.dp)
            ) {
                options.forEach { option ->
                    DropdownMenuItem(
                        text = {
                            Text(
                                text = option,
                                fontWeight = if (option == currentValue) FontWeight.Bold else FontWeight.Normal
                            )
                        },
                        onClick = {
                            onValueChange(option)
                            expanded = false
                        },
                        leadingIcon = {
                            if (option == currentValue) {
                                Icon(
                                    imageVector = Icons.Filled.Check,
                                    contentDescription = null,
                                    tint = PrimaryBlue
                                )
                            }
                        }
                    )
                }
            }
        }

        if (!isLast) {
            Divider(
                color = Color(0xFFF1F5F9),
                modifier = Modifier.padding(vertical = 8.dp)
            )
        }
    }
}